src/pages/Dashboard/MintRepayVai/MintVai/index.tsx

```diff
@@ -0,0 +1,146 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import BigNumber from 'bignumber.js';
+
+import { convertWeiToCoins, convertCoinsToWei } from 'utilities/common';
+import { AmountForm } from 'containers/AmountForm';
+import { AuthContext } from 'context/AuthContext';
+import { SecondaryButton, LabeledInlineContent, TokenTextField } from 'components';
+import { useVaiUser } from 'hooks/useVaiUser';
+import useGetVaiTreasuryPercentage from 'hooks/operations/queries/useGetVaiTreasuryPercentage';
+import useMintVai from 'hooks/operations/mutations/useMintVai';
+import toast from 'components/Basic/Toast';
+import { VAI_SYMBOL } from '../constants';
+import getReadableFeeVai from './getReadableFeeVai';
+import { useStyles } from './styles';
+
+export interface IMintVaiUiProps {
+  disabled: boolean;
+  isMintVaiLoading: boolean;
+  onSubmit: (value: BigNumber) => void;
+  limitWei?: BigNumber;
+  mintFeePercentage?: number;
+}
+
+export const MintVaiUi: React.FC<IMintVaiUiProps> = ({
+  disabled,
+  limitWei,
+  mintFeePercentage,
+  isMintVaiLoading,
+  onSubmit,
+}) => {
+  const styles = useStyles();
+
+  // Convert limit into VAI
+  const readableVaiLimit = React.useMemo(
+    () =>
+      !limitWei
+        ? '-'
+        : convertWeiToCoins({
+            value: limitWei,
+            tokenSymbol: VAI_SYMBOL,
+            returnInReadableFormat: true,
+          }).toString(),
```

Comment:
nit: As a rule of thumb I think ternaries that check if a value is truthy and then return a truthy value on the left and a falsey value on the right are the easiest to read

```suggestion
      limitWei
        ? convertWeiToCoins({
            value: limitWei,
            tokenSymbol: VAI_SYMBOL,
            returnInReadableFormat: true
            : '-',
          }).toString(),
```